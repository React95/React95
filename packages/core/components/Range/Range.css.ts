import { style } from '@vanilla-extract/css';
import { contract } from '../themes/contract.css';
import { calc } from '@vanilla-extract/css-utils';

const track = {
  width: '100%',
  height: contract.space[4],
  backgroundColor: contract.colors.borderDarkest,
  cursor: 'pointer',
  borderStyle: 'solid',
  borderTopWidth: contract.space[1],
  borderTopColor: contract.colors.borderLight,
  borderRightWidth: contract.space[2],
  borderRightColor: contract.colors.borderLightest,
  borderBottomWidth: contract.space[1],
  borderBottomColor: contract.colors.borderLighter,
  borderLeftWidth: contract.space[2],
  borderLeftColor: contract.colors.borderLightest,
  boxShadow: `0px 1px 0px 0px ${contract.colors.borderLightest}`,
} as const;

const thumb = {
  width: contract.space[12],
  height: contract.space[20],
  backgroundColor: contract.colors.material,
  cursor: 'pointer',
  marginTop: calc.negate(contract.space[8]),
  borderStyle: 'solid',
  borderTopWidth: contract.space[1],
  borderTopColor: contract.colors.borderLightest,
  borderRightWidth: contract.space[1],
  borderRightColor: contract.colors.borderDarkest,
  borderBottomWidth: contract.space[1],
  borderBottomColor: contract.colors.borderDarkest,
  borderLeftWidth: contract.space[1],
  borderLeftColor: contract.colors.borderLightest,
  boxShadow: `inset 0px -1px 0px ${contract.colors.borderDark},
  inset -1px 0px 0px ${contract.colors.borderDark},
  inset 0px 1px 0px ${contract.colors.borderLighter},
  inset 1px 0px 0px ${contract.colors.borderLighter}`,
  appearance: 'none',
} as const;

export const range = style({
  appearance: 'none',
  width: '100%',
  selectors: {
    '&::hover,&::focus,&::active': { outline: 'none' },
    '&::-webkit-slider-runnable-track': track,
    '&::-moz-range-track': track,
    '&::-webkit-slider-thumb': thumb,
    '&::-moz-range-thumb': thumb,
    '&:focus::-webkit-slider-runnable-track': {
      backgroundColor: contract.colors.borderDarkest,
    },
  },
});
