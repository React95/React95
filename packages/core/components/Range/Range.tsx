import * as React from 'react';
import { Frame, FrameProps } from '../Frame/Frame';
import { range } from './Range.css';
import cn from 'classnames';

export type RangeProps = Omit<FrameProps<'input'>, 'as'>;

export const Range = React.forwardRef<HTMLInputElement, RangeProps>(
  (rest, ref) => (
    <Frame
      {...rest}
      ref={ref}
      className={cn(range, rest.className)}
      as="input"
      type="range"
    />
  ),
);
