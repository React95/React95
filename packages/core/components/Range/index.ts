import { Range } from './Range';

export default Range;
