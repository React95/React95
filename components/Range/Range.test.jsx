import React from 'react';
import Range from './Range';
import { render } from 'react-testing-library';

describe('<Range />', () => {
  describe('Snapshot', () => {
    it('should match snapshot', () => {
      const { container } = render(<Range />);
      expect(container).toMatchSnapshot();
    });
  });
});
