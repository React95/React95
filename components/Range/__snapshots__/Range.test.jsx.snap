// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Range /> Snapshot should match snapshot 1`] = `
.c0 {
  -webkit-appearance: none;
  width: 100%;
}

.c0:hover,
.c0:focus,
.c0:active {
  outline: none;
}

.c0::-webkit-slider-runnable-track {
  width: 100%;
  height: 4px;
  background: #000;
  cursor: pointer;
  border-top: 1px solid #ccc;
  border-right: 2px solid #fff;
  border-bottom: 1px solid #eee;
  border-left: 2px solid #fff;
  box-shadow: 0px 1px 0px 0px #fff;
}

.c0::-webkit-slider-thumb {
  width: 12px;
  height: 20px;
  background: #c3c7cb;
  cursor: pointer;
  margin-top: -8px;
  border-top: 1px solid #fff;
  border-right: 1px solid #000;
  border-bottom: 1px solid #000;
  border-left: 1px solid #fff;
  box-shadow: inset 0px -1px 0px #aaa,inset -1px 0px 0px #aaa,inset 0px 1px 0px #dedede,inset 1px 0px 0px #dedede;
  -webkit-appearance: none;
}

.c0:focus::-webkit-slider-runnable-track {
  background: #000;
}

.c0::-moz-range-track {
  width: 100%;
  height: 4px;
  background: #000;
  cursor: pointer;
  border-top: 1px solid #ccc;
  border-right: 2px solid #fff;
  border-bottom: 1px solid #eee;
  border-left: 2px solid #fff;
  box-shadow: 0px 1px 0px 0px #fff;
}

.c0::-moz-range-thumb {
  width: 12px;
  height: 20px;
  background: #c3c7cb;
  cursor: pointer;
  margin-top: -8px;
  border-top: 1px solid #fff;
  border-right: 1px solid #000;
  border-bottom: 1px solid #000;
  border-left: 1px solid #fff;
  box-shadow: inset 0px -1px 0px #aaa,inset -1px 0px 0px #aaa,inset 0px 1px 0px #dedede,inset 1px 0px 0px #dedede;
  -webkit-appearance: none;
}

<div>
  <input
    class="c0"
    type="range"
  />
</div>
`;
